File: src/main/java/org/team2471/frc/lib/sensors/Magnepot.java
package org.team2471.frc.lib.sensors;

import edu.wpi.first.wpilibj.AnalogInput;

public class Magnepot extends AnalogInput {

    public Magnepot(int channel) {
        super(channel);
    }

    @Override
    public double pidGet() {
        return (getAverageVoltage() - 2.5) / 2.3 * 180;
    }
}

File: src/main/kotlin/org/team2471/frc/lib/sensors/Magnepot.kt
package org.team2471.frc.lib.sensors

import edu.wpi.first.wpilibj.AnalogInput

class Magnepot(channel: Int) : AnalogInput(channel) {
    override fun pidGet(): Double = (averageVoltage - 2.5) / 2.3 * 180
}
